### PR TITLE
Update Hello dojo to latest dojo v0.6.0 and dojo-starter

### PR DESCRIPTION
**Pull request by gianalarcon**
- Update the tutorial according to the latest dojo `v0.6.0`
- Update the tutorial according to latest `dojo-starter` sample

Todo: 
- Fix current the graphql query 
```graphql
query {
  model(id: "Moves") {
    id
    name
    classHash
    transactionHash
    createdAt
  }
}
```
Use hex get_selector_from _name instead current model raw `name`

### CODE REVIEW

File: docs/pages/cairo/hello-dojo.md [hunk 1]
```diff
@@ -50,7 +51,9 @@ struct Moves {
 
 Notice the `#[derive(Model, Drop, Serde)]` attributes. For a model to be recognized, we _must_ include `Model`. This signals to the Dojo compiler that this struct should be treated as a model.
 
-Our `Moves` model houses a `player` field. At the same time, we have the `#[key]` attribute, it informs Dojo that this model is indexed by the `player` field. If this is unfamiliar to you, we'll clarify its importance later in the chapter. Essentially, it implies that you can query this model using the `player` field. Our `Moves` model also contains the `remaining` and `last_direction` fields
+Next, there's #[dojo::event], a custom attribute enabling the model to be emitted using the emit! macro. This attribute represents one type of Custom Event offered by Dojo.
```
Comment (glihm) [inline docs/pages/cairo/hello-dojo.md:54]:
```suggestion
Next, there's #[dojo::event], a custom attribute enabling the model to be emitted using the `emit!` macro. This attribute represents one type of custom event offered by Dojo.
```

File: docs/pages/cairo/hello-dojo.md [hunk 2]
```diff
@@ -93,20 +96,20 @@ mod actions {
...
+    //     Moved: Moved,
+    // }
+
+    // declaring custom starknet event struct
+    // #[derive(starknet::Event, Model, Copy, Drop, Serde)]
+    // struct Moved {
+    //     #[key]
+    //     player: ContractAddress,
+    //     direction: Direction
+    // }
```
Comment (glihm) [inline docs/pages/cairo/hello-dojo.md:112]: Why is this commented only? We can remove it from here, or did you have something else in mind?

File: docs/pages/cairo/hello-dojo.md [hunk 3]
```diff
@@ -144,19 +143,20 @@ mod actions {
...
             // Calculate the player's next position based on the provided direction.
             let next = next_position(position, direction);
 
-            // // Update the world state with the new moves data and position.
+            // Update the world state with the new moves data and position.
             set!(world, (moves, next));
-
             // Emit an event to the world to notify about the player's move.
-            emit!(world, Moved { player, direction });
+            // emit!(world, Moved { player, direction });
```
Comment (glihm) [inline docs/pages/cairo/hello-dojo.md:158]: Same here, is there a reason to keep this line?

File: docs/pages/cairo/hello-dojo.md [hunk 4]
```diff
@@ -144,19 +143,20 @@ mod actions {
...
+            emit!(world,( Moves { player, remaining, last_direction: direction }));
```
Comment (glihm) [inline docs/pages/cairo/hello-dojo.md:159]: Isn't `moves` already containing all this data actually?

File: docs/pages/cairo/hello-dojo.md [hunk 5]
```diff
@@ -238,13 +233,13 @@ katana --disable-fee
 Success! [Katana](/toolchain/katana/overview.md) should now be running locally on your machine. Now, let's deploy! In your primary terminal, execute:
 
 ```bash
-sozo migrate
+sozo migrate apply
 ```
 
-This command will deploy the artifact to [Katana](/toolchain/katana/overview.md). You should see terminal output similar to this:
+This command will deploy the artifact to `Katana`. You should see terminal output similar to this:
```
Comment (glihm) [inline docs/pages/cairo/hello-dojo.md:239]: We should keep the link I guess, to ensure people can navigate to Katana while reading this. 👍

**Review by glihm — COMMENTED** [review]
Nice work man, here are some comments about this proposal.

File: docs/pages/cairo/hello-dojo.md (hunk 4 above)
Comment (gianalarcon) [reply docs/pages/cairo/hello-dojo.md]: I fixed to emit event just like this `emit!(world, (moves));`. Fixed on commit [24c5edf](https://github.com/dojoengine/book/pull/254/commits/24c5edf6007a0b2e77d3cb4cd472f43aae2aa51c)

**Review by glihm — COMMENTED** [review]
Thanks for the updates @gianalarcon!
One modification to remove some comments in the code, and let's merge. 👍

**Review by glihm — APPROVED** [review]
Thanks for the additional typo fixes. 👍

**Pull request merged**